### PR TITLE
[WIP] Run doctests

### LINKED ISSUE #418: Update docs to indicate refactor of percentile_doy
Refactoring has moved the percentile_doy() function to xclim.core.calendar but a number of examples in the help still indicate imports from utils() 
e.g. hot_days example  (but others present as well): 

>>> import xarray as xr
>>> import xclim.utils
>>> tas = xr.open_dataset("temperature_data.nc").tas
>>> t90 = xclim.utils.percentile_doy(tas, per=0.9)
>>> hot_days = tg90p(tas, t90)

### PR DESCRIPTION
**Pull request by huard**
Run code inside docstrings to make sure examples are in sync with actual code.

Use with `pytest --rootdir tests/ --doctest-modules xclim`

Lots of failing tests. You can add stuff into the namespace using xclim/conftest.py

### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #418 
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [ ] `bumpversion (minor / major / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 

* **Does this PR introduce a breaking change?** 

* **Other information**:

### CODE REVIEW

**Comment by huard** [comment]
I won't have much time to put into this before Friday. @tlogan2000 Is it critical to fix this for the next release ?

File: xclim/indices/_simple.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def tg_max(tas: xarray.DataArray, freq: str = "YS"):
 
     .. math::
 
-        TNx_j = max(TN_{ij})
+        TNx_j = max(TN_{ij}))
```
Comment (Zeitsperre) [inline xclim/indices/_simple.py:63]: Is this a typo?

Comment (huard) [reply xclim/indices/_simple.py:63]: Yes

**Comment by aulemahal** [comment]
I moved all the test configuration to `tests/conftest.py` and one needs to use `--rootdir tests/` in the pytest command so they are read. Also, I fixed the failures of the tests pytest tests (ouf).

However,  I have no idead why pytest doesn't run all the doc tests? Raw `python -m doctest` works as expected, but `pytest` skips a lot of them.

File: xclim/indices/_simple.py (hunk 1 above)
Comment (huard) [inline xclim/indices/_simple.py:63]:
```suggestion
        TNx_j = max(TN_{ij})
```

**Comment by huard** [comment]
@Zeitsperre Will you be able to hook this up in the Travis tests ?

**Comment by aulemahal** [comment]
@Zeitsperre @huard I did add a line in tox.ini to add them to the Travis tests, but did it in the `doc` context, which doesn't have pytest, I forgot...

**Comment by huard** [comment]
I don't see the doctests output in the travis build. I'm trying to fiddle with the config, but don't really know what I'm doing.

**Comment by huard** [comment]
Tests are run inside the xarray master environment to avoid import errors. This would be ready to merge, with the caveat that not all doctests are collected by pytest. I've tried to diagnose why, but no luck so far. I'll open another issue to fix the doctest collection.

**Pull request merged**